app/screens/home/search/search.tsx

```diff
@@ -95,6 +109,18 @@ const SearchScreen = ({teamId}: Props) => {
         setFilter(FileFilters.ALL);
     }, []);
 
+    const updateRecents = useCallback(async () => {
```

Comment:
I don't get why this is needed.

When a search value is removed, this is what should happen:
- The search value is removed from the database
- The observable gets triggered, updating the value of `recentSearches` (the prop)
- The new `recentSearches` is just passed down to the relevant component

That would remove the need of the `recent` state, and the `onRemoveSearch` prop down the tree of components.

Why does that workflow does not work?

Reply:
done.  See latest state of code.